### PR TITLE
[TC-106] Endpoint for returning mock data

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[TC-0106](https://bcdevex.atlassian.net/browse/TC-0106)

Objective:

- returns mock data to the frontend dashboard

### CODE REVIEW

File: backend/src/query-validation.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { ArgumentMetadata, Injectable, PipeTransform } from '@nestjs/common';
+import { Type, plainToInstance } from 'class-transformer';
+import { IsInt, IsString } from 'class-validator';
+
+export class QueryDto {
+  @Type(() => Number)
```
Comment (fw-noel) [inline backend/src/query-validation.pipe.ts:6]: What is @Type achieving?

File: backend/src/common/enums/experience.enum.ts [hunk 2]
```diff
@@ -3,4 +3,11 @@ export enum Experience {
   EXPERIENCED,
   CHIEF_EXPERIENCED,
   OUTSIDE_EXPERIENCED,
-}
\ No newline at end of file
+}
+
+export enum ExperienceName {
+  INTERESTED = 'INTERESTED',
```
Comment (fw-noel) [inline backend/src/common/enums/experience.enum.ts:9]: Going back on what I said on my PR, can you do this `INTERESTED = 'INTERESTED'` in the regular enums, while enumName is meant for user friendly names like `INTERESTED = 'Interested'`?

I was wrong before about the blank enum
`{
INTERESTED,
}`

being the same as definition
`{
INTERESTED = 'INTERESTED',
}`

**Review by fw-noel — APPROVED** [review]
How are you going to determine number of pages on the FE? Random?

**Comment by chelsea-EYDS** [comment]
> How are you going to determine number of pages on the FE? Random?

The front end is sending the requested page + number of rows. Currently I have the mock data setup to create 500 rows, so the number of pages is based on the total number of entries/the number of rows per page.

File: backend/src/common/enums/experience.enum.ts (hunk 2 above)
Comment (chelsea-EYDS) [reply backend/src/common/enums/experience.enum.ts:9]: For sure! I am going to update this on TC-85 if that's okay? (it's pointed to this branch)

File: backend/src/query-validation.pipe.ts (hunk 1 above)
Comment (chelsea-EYDS) [reply backend/src/query-validation.pipe.ts:6]: We are telling class transformer that this should be a number.

**Pull request merged**